addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveAlarmCommandClass.java

```diff
@@ -445,6 +506,14 @@ public Integer getValue() {
                 result.add(getSupportedMessage());
             }
         }
+        if (getVersion() > 2 && pendingEventSupportedReport == false) {
```

Comment:
I think the logic here is still wrong. For example, the refresh parameter is not used except for the V1 class, and if I understand the pendingEventSupportedReport correctly, this just looks like it's true for the duration of the request. So, during the initialisation, it seems that we could get stuck in a perpetual loop if we're unlucky.

Initialize should only return a result the first time it's called (or until the response is fulfilled let's say), or, if refresh is true. It seems that it will actually return a result every time it's called if there's no outstanding request?